features/datachoices/src/main/java/org/opennms/features/datachoices/internal/UsageStatisticsReporter.java

```diff
@@ -298,6 +294,85 @@ private long getDeployedRequisitionWithModifiedFSCount() {
+            installedFeatures = Arrays.stream(m_featuresService.listInstalledFeatures())
+                    .map(f -> f.getName() + "/" + f.getVersion())
+                    .sorted()
+                    .collect(Collectors.joining(","));
+        } catch (Exception e) {
+            installedFeatures = "ERROR: Failed to enumerate the installed features: " + e.getMessage();
+        }
+        return installedFeatures;
+    }
+
+    /**
+     *  This descends the dependency tree breadth-first to find all installed features that have a direct
+     *  or transitive dependence on the opennms-integration-api feature.
+     *
+     *  'opennms-api-layer' is the only core feature that is dependent on the opennms-integration-api.
+     *  Don't include that in the list of installed plugins.
+     */
+    private String getInstalledOIAPluginsByDependencyTree() {
+        List<Feature> featuresDependentOnOIA = new ArrayList<>();
+        try {
+            for (Feature feature : m_featuresService.listInstalledFeatures()) {
+                featuresDependentOnOIA = recurse(feature, featuresDependentOnOIA, List.of("opennms-api-layer"));
+            }
+        }
+        catch (Exception e) {
+            return "ERROR: Failed to enumerate the installed features: " + e.getMessage();
+        }
+        return featuresDependentOnOIA.stream()
+                    .map(f -> f.getName() + "/" + f.getVersion())
+                    .sorted()
+                    .collect(Collectors.joining(","));
+    }
+
+    private List<Feature> recurse(Feature feature, List<Feature> dependencies, List<String> featuresToIgnore) {
+        // should this feature be ignored?
+        if (featuresToIgnore.contains(feature.getName())) {
+            return dependencies;
+        }
+        // Is this feature already known to be dependent on OIA?
+        if (dependencies.contains(feature)) {
+            return dependencies;
+        }
+        // Is this feature directly dependent on OIA?
+        for (Dependency dep : feature.getDependencies()) {
+            if (dep.getName().equals("opennms-integration-api")) {
+                dependencies.add(feature);
+                return dependencies;
+            }
+        }
+        // Are any of this feature's dependencies already known to be dependent on OIA?
+        for (Dependency dep : feature.getDependencies()) {
+            if (dependencies.stream().map(d -> d.getName()).anyMatch(str -> str.equals(dep.getName()))) {
+                dependencies.add(feature);
+                return dependencies;
+            }
+        }
+        // Does this feature have a transitive dependency on OIA that we haven't mapped yet?
+        for (Dependency dep : feature.getDependencies()) {
+            try {
+                int numDependencies = dependencies.size();
+                Feature depFeature = m_featuresService.getFeature(dep.getName());
+                dependencies = recurse(depFeature, dependencies, featuresToIgnore);
```

Comment:
Is there a chance that if  there is circular dependency between two dependencies and this goes into infinite loop ?

Reply:
Does karaf allow circular dependencies? If so, I could keep track of the 'visited' dependencies to avoid that.

Reply:
I'm just concerned on recursive nature here.  We are already checking 2 levels before this routine. That seems enough ?

Reply:
If there were any circular dependencies that don't have an OIA dependency hidden somewhere within, then I think this would loop forever.